exonum/src/events/network.rs

```diff
@@ -337,9 +368,19 @@ impl NetworkHandler {
                                 socket.get_ref().peer_addr().unwrap(),
                             );
                             pool.add(&key, conn_addr.clone(), sender_tx);
-                            let connection =
-                                Connection::new(handle, socket, receiver_rx, conn_addr);
-                            to_box(Self::handle_connection(connection, message, &network_tx))
+                            let connection = Connection::new(
+                                handle,
+                                socket,
+                                receiver_rx,
+                                conn_addr,
+                                *message.pub_key(),
+                            );
+                            to_box(Self::handle_connection(
+                                connection,
+                                message,
+                                pool,
+                                &network_tx,
+                            ))
```

Comment:
All these
[ｖ
ｅ
ｒ
ｔ
ｉ
ｃ
ａ
ｌ](https://webimg.ccpgamescdn.com/zncvdb47ti8o/45K7khNcBaKMsyC8SCMGES/a88230d4ce62b115139ed9a38d21d58b/EVEOnline_ships_Naglfar.jpg)
argument lists everywhere leave me with a feeling that some sort of a _context_ could be of use here. So that we can bundle these things and pass them around as a whole. I don't know, we could put them in `self` or some explicit struct.

This is not a nit. You don't have to change anything in this PR. The situation is far from dire in this file. That's just a thing that we might think about in background. If there is some natural, obvious way to group these values then we might use it.